neqo-transport/src/stream_id.rs

```diff
@@ -127,52 +105,172 @@ impl ::std::fmt::Display for StreamId {
     }
 }
 
-#[derive(Debug, Eq, PartialEq, Clone, Copy, Ord, PartialOrd, Hash)]
-pub struct StreamIndex(u64);
-
-impl StreamIndex {
-    pub const fn new(val: u64) -> Self {
-        Self(val)
-    }
+pub struct RemoteStreamsFlowControl {
+    streams_fc: ReceiverFlowControl<StreamType>,
+    next_stream: u64,
+    stream_type_bits: u64,
+}
 
-    pub fn to_stream_id(self, stream_type: StreamType, role: Role) -> StreamId {
+impl RemoteStreamsFlowControl {
+    pub fn new(stream_type: StreamType, max_streams: u64, role: Role) -> Self {
         let type_val = match stream_type {
             StreamType::BiDi => 0,
             StreamType::UniDi => 2,
         };
+        // This is for a stream created by a peer, therefore the role bit is opposit.
         let role_val = match role {
-            Role::Server => 1,
-            Role::Client => 0,
+            Role::Server => 0,
+            Role::Client => 1,
         };
 
-        StreamId::from((self.0 << 2) + type_val + role_val)
+        Self {
+            streams_fc: ReceiverFlowControl::new(stream_type, max_streams),
+            next_stream: 0,
+            stream_type_bits: type_val + role_val,
+        }
     }
 
-    pub fn as_u64(self) -> u64 {
-        self.0
+    pub fn write_frames(
```

Comment:
There are lots of functions that forward to `self.fc`.  Maybe you can implement Deref and avoid the extra boilerplate.